### ISSUE TITLE
Any feature to access loss value in mxnet like keras package for tensorflow

### ISSUE THREAD

**Issue by lincolnBush** [apache/incubator-mxnet#5994]
In keras optimizers.py:

def get_updates(self, params, constraints, loss):

while in mxnet we only have weight and grad in update function:
Is it convenient to change mxnet to have such feature

**Comment by kevinthesun** [comment]
Do you mean you would like to access loss inside optimizer update method?

**Comment by lincolnBush** [comment]
@kevinthesun  Yeah, do you get any ideas

**Comment by kevinthesun** [comment]
```
    def update(self):
        """Update parameters according to the installed optimizer and the gradients computed
        in the previous forward-backward batch.
        """
        assert self.binded and self.params_initialized and self.optimizer_initialized

        self._params_dirty = True
        if self._update_on_kvstore:
            _update_params_on_kvstore(self._exec_group.param_arrays,
                                      self._exec_group.grad_arrays,
                                      self._kvstore)
        else:
            _update_params(self._exec_group.param_arrays,
                           self._exec_group.grad_arrays,
                           updater=self._updater,
                           num_device=len(self._context),
                           kvstore=self._kvstore)
```
This is update method of module. You can override it to do extra work when updating.

**Comment by lincolnBush** [comment]
Thank you very much. But this still seems hard to solve. Because there seems to be no interface to access loss value only grad is provided. It maybe needed to change some c++ code.

**Comment by kevinthesun** [comment]
Does this issue still exists? Can we close it?
